engine/src/flutter/shell/platform/windows/flutter_windows_internal.h

```diff
@@ -65,6 +65,15 @@ FLUTTER_EXPORT void FlutterDesktopEngineRegisterPlatformViewType(
     const char* view_type_name,
     FlutterPlatformViewTypeEntry view_type);
 
+// Returns the engine associated with the given identifier. Engine identifier
+// must be valid and for a running engine, otherwise the behavior is undefined.
+//
+// Identifier can be obtained from PlatformDispatcher.instance.engineId.
+//
+// This method must be called from the platform thread.
+FLUTTER_EXPORT FlutterDesktopEngineRef FlutterDesktopEngineForId(
+    int64_t engine_id);
```

Comment:
Is it intentional that this is on the internal header file?

This is perfectly reasonable if you are concerned this API isn't production-ready yet... but I think it'd be OK to go straight to the public header file: `engine/src/flutter/shell/platform/windows/public/flutter_windows.h`.

Feel free to keep as-is, I'm just double-checking this was intentional :)

Reply:
@stuartmorgan is concerned that having this in public API opens a  can of worms, mostly if people start using bulk of the engine API from FFI, in which case we don't have good way to migrate the API without relying on runtime errors. You can read the whole conversion in attached issue. So for the time being the API stays private. It is still very useful to have this as we can start experiment with dogfooding FFI inside embedders.